ui/src/treeDataProvider/AzureParentNode.ts

```diff
@@ -75,21 +75,33 @@ export class AzureParentNode<T extends IAzureParentTreeItem = IAzureParentTreeIt
     }
 
     public async loadMoreChildren(): Promise<void> {
-        let clearCache: boolean = false;
-        if (this._cachedChildren === undefined) {
+        if (this._loadMoreChildrenTask) {
+            await this._loadMoreChildrenTask;
```

Comment:
1. On a first read, it looks like line 84's `finally` obviates this branch. 
Is this a guard against the race condition, where `loadMoreChildren` is called again before the `finally` is executed? 

2. line 83 (identical) has error catching while this line does not. Can you fill me in on what I'm missing here?

Reply:
1. Yes. loadMoreChildren can be called multiple times at once and the first block of the if statement will only be entered if two calls are happening at the same time
2. See other comment. It has a `finally` block, not a `catch` block. There is no error handling.